plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.ts

```diff
@@ -129,60 +136,73 @@ export class OktaGroupEntityProvider extends OktaEntityProvider {
       logger: this.logger,
     });
 
-    await Promise.allSettled(
-      Object.entries(oktaGroups).map(async ([_, group]) => {
-        const members: string[] = [];
-        await group.listUsers().each(user => {
+    for (const entries of chunk(Object.entries(oktaGroups), this.chunkSize)) {
```

Comment:
I feel like p-map might be a good choice for this type of thing.. but Im not sure if it has the allSettled type behaviour.